src/main/java/medic/gateway/alert/WakefulService.java

```diff
@@ -43,20 +47,22 @@ public void doWakefulWork(Intent intent) {
 
 		try {
 			WebappPoller poller;
-			do {
 
+			while (keepPollingWebapp) {
```

Comment:
I think the `do while` format was more readable than storing a boolean and using `while`. Why did you switch this over?

Reply:
I thought `while` was more readable haha, `do while` looks bit old to me. 
But originally I changed it to `while` because I don't want to cache `lastResponse`, I wanted to reuse the one available inside the `while` by passing it to the method that determines if the loop continues or exit. Then I needed a variable outside the loop to keep the result of this method as bool.. then It made more sense to me to switch it to `while`.